### PR TITLE
probes: use veth device for probing managed neighbor support

### PR DESCRIPTION
**Pull request by ti-mo**
Some distributions don't ship with dummy device support/module, leading to the following error when probing for managed neighbors:

```
level=fatal msg="failed to probe managed neighbor support"
error="failed to add dummy link: operation not supported" subsys=probes
```

This patch uses a veth device instead, which should be available on nodes running Cilium, if it's not already loaded.

```release-note
Use veth device for probing managed neighbor support
```

### CODE REVIEW

**Review by rgo3 — APPROVED** [review]

**Comment by ti-mo** [comment]
/test-1.26-net-next

File: pkg/datapath/linux/probes/managed_neighbors.go [hunk 1]
```diff
@@ -70,17 +70,22 @@ func haveManagedNeighbors() (outer error) {
...
 	}()
 
-	la := netlink.NewLinkAttrs()
-	la.Name = "cilium-dummy"
+	// Use a veth device instead of a dummy to avoid the kernel having to modprobe
+	// the dummy kmod, which could potentially be compiled out. veth is currently
+	// a hard dependency for Cilium, so safe to assume the module is available if
+	// not already loaded.
+	veth := &netlink.Veth{
+		LinkAttrs: netlink.LinkAttrs{Name: "veth0"},
```
Comment (squeed) [inline pkg/datapath/linux/probes/managed_neighbors.go:78]: are you worried about name collisions? what about `cilium-dummy-veth0`?

Comment (ti-mo) [reply pkg/datapath/linux/probes/managed_neighbors.go:78]: This runs in a new netns, so no.

**Comment by ti-mo** [comment]
/test-1.26-net-next

**Pull request merged**